### ISSUE TITLE
TypeError: string indices must be integers in selenium-py 2.49.1

### ISSUE THREAD

**Issue by nikolas** [SeleniumHQ/selenium#1497]
When upgrading from selenium 2.48.0 to selenium 2.49.1, I'm getting this error:

```
Traceback (most recent call last):
        File "/home/travis/build/ccnmtl/worth2/ve/lib/python2.7/site-packages/behave/model.py", line 1456, in run
          match.run(runner.context)
        File "/home/travis/build/ccnmtl/worth2/ve/lib/python2.7/site-packages/behave/model.py", line 1903, in run
          self.func(context, *args, **kwargs)
        File "features/steps/common.py", line 67, in i_click_the_link
          element.click()
        File "/home/travis/build/ccnmtl/worth2/ve/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 75, in click
          self._execute(Command.CLICK_ELEMENT)
        File "/home/travis/build/ccnmtl/worth2/ve/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 454, in _execute
          return self._parent.execute(command, params)
        File "/home/travis/build/ccnmtl/worth2/ve/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 201, in execute
          self.error_handler.check_response(response)
        File "/home/travis/build/ccnmtl/worth2/ve/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 107, in check_response
          message = value["value"]["message"]
      TypeError: string indices must be integers
```

Here's the failing travis build:
https://travis-ci.org/ccnmtl/worth2/builds/103229032#L1113

And here's an example of the same exception being raised in another Django application:
https://travis-ci.org/ccnmtl/uelc/builds/103229165#L1938

**Comment by xavfernandez** [comment]
Same thing with `selenium==2.49.2`:

    Traceback (most recent call last):
      File "/home/xfernandez/dev/project/project/browser_tests.py", line 235, in test_check_tasks
        self.driver.find_element_by_xpath("//button/span[contains(., 'Send')]").click()
      File "/home/xfernandez/.virtualenvs/project/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 75, in click
        self._execute(Command.CLICK_ELEMENT)
      File "/home/xfernandez/.virtualenvs/project/local/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 454, in _execute
        return self._parent.execute(command, params)
      File "/home/xfernandez/.virtualenvs/project/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 201, in execute
        self.error_handler.check_response(response)
      File "/home/xfernandez/.virtualenvs/project/local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 107, in check_response
        message = value["value"]["message"]
    TypeError: string indices must be integers

**Comment by AutomatedTester** [comment]
Please can you create a reduced test case so I don't need to get your whole repo out and see how you build things.

**Comment by nikolas** [comment]
I'm not familiar enough with Selenium's internals to create a useful test case. However, it seems that the problem is, at the code here: https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/errorhandler.py#L107 selenium expects `value` to be JSON, not a string. We just need to handle the case where it's a string, not a JSON object. In order to do that, my inclination would be to refactor that code a bit since it's already pretty complex. I might come up with a PR for this if no one else does.

**Comment by AutomatedTester** [comment]
a reduced test case is the html needed and a quick test script (this
includes what browser and everything that might be useful) to do this.

If it is not sending over JSON there is some funkiness going on.

Your bug report doesn't actually give anything meaningful to be able to
action it except a log.

**Comment by nikolas** [comment]
Alright, I'll see if I can narrow down where the problem is to make a test case.

**Comment by hmcelroy** [comment]
I'm also getting this error "TypeError: string indices must be integers" on 24 tests after updating selenium after updating firefox (43.0.4).  Attached is my test results - you can see in all instances of the type error, there is a click that is setting it off.  
[test failures gpss-238 en.txt](https://github.com/SeleniumHQ/selenium/files/98628/test.failures.gpss-238.en.txt)

**Comment by francoisfreitag** [comment]
I've ran into the same problem, here is a stripped down version of the problem:
``` html
<!doctype html>
<html>
<body>
    <h1>Issue 1497</h1>
    <button id="button1">
        <span id="nested-span">Click to reproduce</span>
    </button>
</body>
</html>
```

``` python
def test_issue_1497(self):
     self.driver.get('file:///path/to/dev/test.html')
     self.driver.find_element_by_css_selector('#span1').click()
```

```
Traceback (most recent call last):
  File "/path/to/test.py", line 6, in test_issue_1497
    self.driver.find_element_by_css_selector('#span1').click()
  File "/selenium/webdriver/remote/webdriver.py", line 402, in find_element_by_css_selector
    return self.find_element(by=By.CSS_SELECTOR, value=css_selector)
  File "selenium/webdriver/remote/webdriver.py", line 712, in find_element
    {'using': by, 'value': value})['value']
  File "selenium/webdriver/remote/webdriver.py", line 201, in execute
    self.error_handler.check_response(response)
  File "selenium/webdriver/remote/errorhandler.py", line 181, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: {"method":"css selector","selector":"#span1"}
Stacktrace:
    at FirefoxDriver.prototype.findElementInternal_ (file:///tmp/tmp_rsPXa/extensions/fxdriver@googlecode.com/components/driver-component.js:10675)
    at FirefoxDriver.prototype.findElement (file:///tmp/tmp_rsPXa/extensions/fxdriver@googlecode.com/components/driver-component.js:10684)
    at DelayedCommand.prototype.executeInternal_/h (file:///tmp/tmp_rsPXa/extensions/fxdriver@googlecode.com/components/command-processor.js:12551)
    at DelayedCommand.prototype.executeInternal_ (file:///tmp/tmp_rsPXa/extensions/fxdriver@googlecode.com/components/command-processor.js:12556)
    at DelayedCommand.prototype.execute/< (file:///tmp/tmp_rsPXa/extensions/fxdriver@googlecode.com/components/command-processor.js:12498)
```

The workaround for me was to select the button element, not the span that's inside.

**Comment by sumenon** [comment]
I am facing this very same issue.Waiting for a fix.

**Comment by nikolas** [comment]
Here is a workaround: https://github.com/SeleniumHQ/selenium/pull/1547

Let me know if it works.

**Comment by ernestm** [comment]
I'm having this issue too, glad to see there's a fix in flight...

**Comment by barancev** [comment]
Please retest in 2.50

**Comment by francoisfreitag** [comment]
2.50.0 does not fix the issue.

``` shell
pip freeze | grep selenium                                                            
selenium==2.50.0
```

```
ERROR: test_issue_1497
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/path/to/tests.py", line 178, in test_issue_1497
    self.driver.find_element_by_css_selector('#span1').click()
  File "selenium/webdriver/remote/webelement.py", line 75, in click
    self._execute(Command.CLICK_ELEMENT)
  File "selenium/webdriver/remote/webelement.py", line 460, in _execute
    return self._parent.execute(command, params)
  File "selenium/webdriver/remote/webdriver.py", line 201, in execute
    self.error_handler.check_response(response)
  File "selenium/webdriver/remote/errorhandler.py", line 107, in check_response
    message = value["value"]["message"]
TypeError: string indices must be integers
```

**Comment by kurkha** [comment]
I'm also still having this issue in 2.50 with Firefox (at least). I have the following structure:

```
<div class="ui-dialog-buttonset">
    <button aria-disabled="false" role="button" class="ui-button ui-widget ui-state-default ui-corner-all  
    ui-button-text-only" type="button">
        <span class="ui-button-text">Ok</span>
    </button>
```
Now, I get the error by using the following command (It's Robot code, but the traceback is identical for the Selenium-related part):

`Click Element    xpath=//button/span[@class='ui-button-text' and contains(., 'Ok')]`

Again, 2.48 works fine.

**Comment by barancev** [comment]
Fixed by commit c68ace0
